pkg/store/proxy.go

```diff
@@ -49,12 +67,41 @@ type ProxyStore struct {
 	selectorLabels labels.Labels
 
 	responseTimeout time.Duration
+	metrics         *proxyStoreMetrics
+}
+
+func newProxyStoreMetrics(reg *prometheus.Registry) *proxyStoreMetrics {
+	var m proxyStoreMetrics
+
+	m.timeToFirstByte = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "thanos_proxy_time_to_first_byte_seconds",
+		Help:    "Time to get 1st frame from store.",
+		Buckets: []float64{0.01, 0.1, 0.3, 0.6, 1, 3, 6, 9, 20, 30, 60, 90, 120, 240, 360, 720},
+	}, []string{"external_labels", "store_type", "payload"})
+
+	m.frameTimeoutCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_proxy_frame_timeouted_requests_count",
+		Help: "Number of requests with expired store.response-timeout",
+	}, []string{"external_labels", "store_type"})
```

Comment:
Not sure if having `external_labels` is good idea as it might be a huge string with multiple key value pairs. I don't think anyone would query `thanos_proxy_frame_timeouted_requests_count{external_labels="a=b c=d"}` ?

Reply:
Look for https://github.com/thanos-io/thanos/pull/1789#discussion_r351232976 discuss please )

Reply:
Yep, and we have rejected PRs with these labels in the past because our users could be using service discovery mechanisms with highly dynamic i.e. temporary external labels.